### PR TITLE
Docs: Update and fix various markdown issues in the docs

### PR DESCRIPTION
**Pull request by Risae**
*Issue #, if available:*

\-

*Description of changes:*

Various rain docs are broken, in this PR i tried to fix them and add some missing information. See diff for changed made to the docs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ericzbeard** [comment]
I appreciate the effort, but these files are auto-generated. To fix issues, you will need to edit the source and regenerate them.

```sh
go generate ./...
```

**Comment by Risae** [comment]
> I appreciate the effort, but these files are auto-generated. To fix issues, you will need to edit the source and regenerate them.
> 
> ```shell
> go generate ./...
> ```

Should be ready for review now.

**Comment by Risae** [comment]
I just noticed that 

```
docs/rain_check.md
docs/rain_version.md
```

is not auto-generated. Is this working as intended?

File: internal/cmd/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ var force = false
 var Cmd = &cobra.Command{
 	Use:                   "bootstrap",
 	Short:                 "Creates the artifacts bucket",
-	Long:                  "Creates a s3 bucket to hold all the artifacts generated and referenced by rain cli",
+	Long:                  `Creates an S3 Bucket to hold all the artifacts generated and referenced by rain cli.`,
```
Comment (ericzbeard) [inline internal/cmd/bootstrap/bootstrap.go:14]: Why backticks here?

File: internal/cmd/deploy/deploy.go [hunk 2]
```diff
@@ -29,18 +29,19 @@ var ignoreUnknownParams bool
...
-The bucket's name will be of the format rain-artifacts-<AWS account id>-<AWS region>.
+The bucket's name will be of the format "rain-artifacts-\<AWS account id\>-\<AWS region\>".
 
 The config flag can be used to programmatically set tags and parameters.
 The format is similar to the "Template configuration file" for AWS CodePipeline just without the
-'StackPolicy' key. The file can be in YAML or JSON format.
+"StackPolicy" key. The file can be in YAML or JSON format.
 
 JSON:
+~~~json
```
Comment (ericzbeard) [inline internal/cmd/deploy/deploy.go:44]: I normally use backticks for GitHub markdown, have you confirmed that this renders correctly?

**Comment by ericzbeard** [comment]
Unit tests are failing. The output needs to work for both the CLI and markdown. There are tests that confirm CLI output.

File: internal/cmd/bootstrap/bootstrap.go (hunk 1 above)
Comment (Risae) [reply internal/cmd/bootstrap/bootstrap.go:14]: To keep it consistent with the other code. Only with backticks you are able to use `\<` without things breaking while using `go generate`

**Comment by Risae** [comment]
> Unit tests are failing. The output needs to work for both the CLI and markdown. There are tests that confirm CLI output.

That is certainly bad. To be honest i don't know how to fix that. If you want to write proper markdown, the CLI will break. If you want to write proper CLI, the markdown with break.
Only solution, that i can think of, is to create another pipeline which "fixes" the broken markdown docs. Here is current (broken) version of the documentation that i tried to fix: https://aws-cloudformation.github.io/rain/rain_deploy.html

See for example `rain-artifacts--`.

**Comment by ericzbeard** [comment]
I'll have to dig into the Cobra docs to see how we can customize the output. Seems like a miss to not account for markdown formatting. The docs do need to be auto-generated, though. Otherwise it would be very difficult to keep them in sync.

**Pull request closed without merging**